src/HTMLGenerator/testSuite/sequencer.py

```diff
@@ -3,18 +3,76 @@
 import pathlib
 import errno
 import json
+import time
 from pathlib import Path
 from asserter import TestCase
 from tqdm import tqdm
 import utils
+import subprocess
 from utils import *
 import sys
 sys.path.append("..")
-import converter
-from converter import convert
 
+CURRENT_CONFIGURATION_PATH = "currentConfiguration.txt"
+NOTIFICATION_PATH = "complete.txt"
+LABVIEW_PATH = Path("C:/Program Files/National Instruments/LabVIEW 2017/LabVIEW.exe")
 ERROR_REPORT_JSON = r'errorReport.json'
 
+def _get_path_relative_to_username(filename):
+    # Return a path that is relative to the ~user
+    s = Path("~") 
+    filename_path = Path(filename)
+    return str(filename_path.relative_to(s.expanduser()))
+
+def convert_test_suite(top_level_output_directory, top_level_input_directory, json_with_filepaths, path_to_current_working_directory = Path.cwd()):
+    converter_path = Path(path_to_current_working_directory / "ConvertFromConfigurationFile.vi")
+    current_configuration_absolute_path = Path(path_to_current_working_directory / CURRENT_CONFIGURATION_PATH)
+    notification_absolute_path = Path(path_to_current_working_directory / NOTIFICATION_PATH)
+    call_converter_command = str(LABVIEW_PATH) + " " + str(converter_path)
+
+    try:
+        if os.path.isfile(notification_absolute_path):
+            os.remove(notification_absolute_path)                
+    except:
+        pass
+    # Writing the top, preload, input and output path to the currentConfiguration.txt file.
+    try:
+        with open(current_configuration_absolute_path, 'w') as current_configuration_file:
+            current_configuration_file.write(
+                '/top:' + str(top_level_output_directory / json_with_filepaths["topPath"]) + "\n")
+            current_configuration_file.write(
+                '/preload:' + str(Path(json_with_filepaths["preloadedFiles"])) + "\n")
+            current_configuration_file.write(
+                '/input:' + str(top_level_input_directory / json_with_filepaths["inputPath"]) + "\n")
+            current_configuration_file.write(
+                '/out:' + str(Path(json_with_filepaths["outputPath"])) + "\n")
+            current_configuration_file.flush()
+
+    except FileNotFoundError as f:
+        error_message = str(f.strerror, _get_path_relative_to_username(f.filename))
+        return error_message
+
+    else:
+        # convert(path_to_current_working_directory)
+        try:
+            process = subprocess.Popen(call_converter_command, stdin=subprocess.PIPE)
+        except OSError as e:
+            error_message = str(e)
+            print(error_message , "\n")
+            return error_message   
+
+    while(True):
+        if(os.path.isfile(notification_absolute_path)): 
+            process.stdin.close()
+            process.wait()
+            break
+        else:
+            time.sleep(3)
+    os.remove(notification_absolute_path)
+    os.remove(current_configuration_absolute_path)
```

Comment:
All this is the exact same code as the original converter, right? Can that not be reused?

Basically, the only thing that's different in the test suite is the way the config file is written, right? So that should be the only thing function that should have a different implementation compared to the converter. The remaining part, which is launching LabVIEW to run the converter VI should be reusable.

The converter should not be aware of the test suite, but the test suite should be able to reuse the converter without copy-pasting code from it.

Reply:
Okay I will reuse the run_converter()